source/routing/raptor_api.cpp

```diff
@@ -76,15 +78,14 @@ static void fill_shape(pbnavitia::Section* pb_section,
                 prev_coord = cur_coord;
             }
         }
-        // Add the coordinates of the stop point if not already added
-        // by the shape.
-        const auto& sp_coord = st->stop_point->coord;
-        if (sp_coord != prev_coord) {
-            add_coord(sp_coord, pb_section);
-            prev_coord = sp_coord;
-        }
```

Comment:
**Personally**, I'm against this change. It breaks the invariant "the shape passes by all the intermediate stops".

Reply:
Should a shape really pass by all the intermediate stops? (I don't have the answer really)

From our point of view, and with the data we have to work with, the people creating shapes of our lines usually draw on the corresponding roads on Here or OSM datas. And if a road is let's say 5 or 6 meter wide it's usually quite narrow on the resulting tiles. The shape is more on the middle of the road then.
Our stops are usually at the precise coordinate of the physical post, which is on a sidewalk, and not at all matched with an OSM road.
That's what is causing the bits of shape if the first screenshot from @vpassama. Our stops and shapes are usually a few meters apart. It may be really specific to us though.

The only thing I found to maybe support our case is the gtfs specification https://developers.google.com/transit/gtfs/reference/?hl=en#shapestxt and the last sentence:
> Shapes describe the physical path that a vehicle takes, and are defined in the file shapes.txt. Shapes belong to Trips, and consist of a sequence of points. Tracing the points in order provides the path of the vehicle. The points do not need to match stop locations.

Some people seems against this sentence and it lead to interesting discussions : https://github.com/google/transit/pull/39

Reply:
I do not have a strong opinion on this.
Both approach have their downside: if you have clean shape forcing them to pass by each stop point make them ugly, but if your shape are shitty from the start making them pass by the stop point make them at least understable by the user.

Reply:
Same, I'm not UX/UI guy, no strong opinion.
Maybe we can parametrize that... Not sure it's simple, though.